scripts/main.js

```diff
@@ -1,66 +1,217 @@
+        this.$genreSubmitter = document.getElementById(genreSubmitter);
+        this.genreHandler = genreHandler;
     }
 
-    createMovieEl(movie){
-        const {image, title, genre, year,id} = movie;
-        return `<tr data-id="${id}"><td><img src="${image}"></td><td>${title}</td><td>${genre}</td><td>${year}</td></tr>`
+    createMovieEl(movie) {
+        const {
+            image,
+            title,
+            genre,
+            year,
+            id
+        } = movie;
+        return `<tr data-id="${id}" data-year="${year}"><td><img src="${image}"></td><td>${title}</td><td>${genre}</td><td>${year}</td></tr>`;
     }
 
-    fillTable(){
+    //Please Open with Live Server
+    //This homework is a production of the collaboration with İrem Kenar
+    
+    //To create year radio buttons dynamically
+    createYearRadioBtn(year, yearCount) {
+        return `<div class="form-check">
+            <input class="form-check-input" type="radio" name="year" id = ${year}  value=${year}>
+               <label class="form-check-label" for=${year}>
+                ${year} (${yearCount})
+              </label>
+               </div>`;
+    }
+    //To insert year with number to radio buttons on html
+
+    addToRadioBtn(content) {
+        const $el = document.getElementById("filter-by-year");
+        //
+        $el.insertAdjacentHTML("afterbegin", content);
+    }
+
+
+    //To create genre check buttons dynamically
+    createGenreCheckBtn(genre, genreCount) {
+        return `<div class="form-check">
+             <input class="form-check-input" type="checkbox" id="genre-${genre}" name="genre" value=${genre}>
+               <label class="form-check-label" for="genre-${genre}">
+                ${genre} (${genreCount})
+                </label>
+    </div>`;
+    }
+
+    // To insert genre with number check buttons on html
+    addToCheckBtn(content) {
+        const $el = document.getElementById("filter-by-genre");
+        $el.insertAdjacentHTML("afterbegin", content);
+    }
+
+    //To create a movie object that includes only years and genre from data.js
+    createMovieObjects() {
+        const movieObj = {
+            sortByYear : {},
+            sortByGenre : {}
+        };
+        return movieObj;
+    }
+
+    //To bind events for creation and insertion of radio and check buttons to html
+    putBtnsInHtml() {
+        const movieObj = this.createMovieObjects();
+
+        for (let i = 0; i < data.length; i++) {
+            if (!movieObj.sortByYear[data[i].year]) {
+                movieObj.sortByYear[data[i].year] = 1;
+            } else {
+                movieObj.sortByYear[data[i].year]++;
+            }
+
+            if (!movieObj.sortByGenre[data[i].genre]) {
+                movieObj.sortByGenre[data[i].genre] = 1;
+            } else {
+                movieObj.sortByGenre[data[i].genre]++;
+            }
+        }
+
+        console.log(movieObj);
+
+        for (let key in movieObj.sortByYear) {
```

Comment:
for donguleri yerine arrayin foreach metodyla halledebilir miydik?

Reply:
Halledebiliriz sanırım hocam boş bir obje olusturup istenenleri for döngüsüyle atmak yerine, boş iki dizi oluşturup data arrayi üzerinden oluşturduğumuz boş ikiye diziye forEach metoduyla ayrı ayrı gerekli işlemleri yapıp aktarabiliriz. Yine data arrayi üzerinden reduce, map kullanarakta metodların şartlarına uygun olarakta halledebiliriz. Doğru mu düşündüm acaba hocam teşekkür ederim.